packages/compass-workspaces/src/stores/workspaces.ts

```diff
@@ -318,10 +318,23 @@ const reducer: Reducer<WorkspacesState> = (
           : state;
       }
 
-      // ... otherwise check if we can replace the current tab based on its
-      // replace handlers and force new tab opening if we can't
       if (currentActiveTab) {
-        forceNewTab = canReplaceTab(currentActiveTab) === false;
+        // if both the new workspace and the existing one are connection scoped,
+        // make sure we do not replace tabs between different connections
+        if (
+          action.workspace.type !== 'Welcome' &&
```

Comment:
Do we need to enumerate the types or this can be checked with `currentActiveTab.connectionId && action.workspace.connectionId`?

Reply:
Typescript does not allow to reference the property, because some of the workspaces don't have it. If we really wanted to avoid this, we could add `connectionId?: never` to the global workspaces, but I'd rather enumerate like this

this is not valid:
```
type MyType = { type: 'A' } | { type: 'B', prop: string };

const hasProp = (obj: MyType) => !!obj.prop;
```

this is valid:
```
type MyType = { type: 'A', prop?: never } | { type: 'B', prop: string };

const hasProp = (obj: MyType) => !!obj.prop;
```